### PR TITLE
Adapt pika to use the p2300 spack package

### PR DESCRIPTION
**Pull request by aurianer**

### CODE REVIEW

File: var/spack/repos/builtin/packages/pika/package.py [hunk 1]
```diff
@@ -59,6 +59,8 @@ class Pika(CMakePackage, CudaPackage, ROCmPackage):
     variant("mpi", default=False, description="Enable MPI support")
     variant("apex", default=False, description="Enable APEX support", when="@0.2:")
     variant("tracy", default=False, description="Enable Tracy support", when="@0.7:")
+    variant("p2300", default=False, description="Use P2300 reference \
+            implementation for sender/receiver functionality", when="@0.8:")
```
Comment (msimberg) [inline var/spack/repos/builtin/packages/pika/package.py]:
```suggestion
            implementation for sender/receiver functionality", when="@main")
```

File: var/spack/repos/builtin/packages/pika/package.py [hunk 2]
```diff
@@ -89,9 +92,11 @@ class Pika(CMakePackage, CudaPackage, ROCmPackage):
     depends_on("rocblas", when="+rocm")
     depends_on("hipblas", when="+rocm")
     depends_on("rocsolver", when="@0.5: +rocm")
+    depends_on("p2300", when="@0.8: +p2300")
 
     for cxxstd in cxxstds:
-        depends_on("boost cxxstd={0}".format(map_cxxstd(cxxstd)), when="cxxstd={0}".format(cxxstd))
+        depends_on("boost cxxstd={0}".format(map_cxxstd(cxxstd)),
+                when="cxxstd={0}".format(cxxstd))
```
Comment (msimberg) [inline var/spack/repos/builtin/packages/pika/package.py]: Are the formatting changes necessary for CI to pass?

File: var/spack/repos/builtin/packages/pika/package.py [hunk 3]
```diff
@@ -89,9 +92,11 @@ class Pika(CMakePackage, CudaPackage, ROCmPackage):
     depends_on("rocblas", when="+rocm")
     depends_on("hipblas", when="+rocm")
     depends_on("rocsolver", when="@0.5: +rocm")
+    depends_on("p2300", when="@0.8: +p2300")
```
Comment (msimberg) [inline var/spack/repos/builtin/packages/pika/package.py]:
```suggestion
    depends_on("p2300", when="+p2300")
```

**Review by msimberg — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/pika/package.py (hunk 2 above)
Comment (aurianer) [reply var/spack/repos/builtin/packages/pika/package.py]: yes looks like, I will update

**Comment by msimberg** [comment]
This looks good to me besides the formatting changes required by CI. However, I'm just waiting for us to merge https://github.com/pika-org/pika/pull/436 before actually approving this since it's required for this to actually work.

**Review by teonnik — DISMISSED** [review]

**Review by teonnik — DISMISSED** [review]

**Comment by aurianer** [comment]
https://github.com/pika-org/pika/pull/436 has been merged, feel free to merge when ci has passed

**Review by msimberg — DISMISSED** [review]

**Comment by msimberg** [comment]
Is there anything required on this PR to get `waiting for style check to succeed` to finish?

**Comment by aurianer** [comment]
I force pushed as the CI seemed to be stuck

**Review by haampie — APPROVED** [review]

**Pull request merged**